### ISSUE TITLE
cloud_to_cloud:c++:http2_server:rst_after_data failure

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#10624]
possibly related to https://github.com/grpc/grpc/issues/10623, cc @ericgribkoff if there's an issue in `http2_server`

```
Error Message

Failure
Standard Output

E0413 00:59:20.555743904       1 http2_client.cc:111]        assertion failed: response.has_payload()
```

log https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/13226/#showFailuresLink

seen in pr https://github.com/grpc/grpc/pull/10616

**Comment by jtattermusch** [comment]
Saw similar issue for java client. https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/13942/

**Comment by yang-g** [comment]
I think what you saw should be #10455 (or a new issue).
This one (C++) should have been fixed by #10775 
Closing
